src/syntax-highlighters/monaco/providers/completion.ts

```diff
@@ -0,0 +1,406 @@
+            label: "define-from-object",
+            kind: monaco.languages.CompletionItemKind.Function,
+            // eslint-disable-next-line no-template-curly-in-string
+            insertText: "define ${1:relation} as ${2:relation_in_other_object} from ${2:another_relation}",
+            insertTextRules: monaco.languages.CompletionItemInsertTextRule.InsertAsSnippet,
+            range,
+          },
+          {
+            label: "define",
+            kind: monaco.languages.CompletionItemKind.Keyword,
+            insertText: "define",
+            range,
+          },
+        ],
+      };
+    }
+
+    return {
+      suggestions: [
+        {
+          label: "sample-gdrive",
+          kind: monaco.languages.CompletionItemKind.Keyword,
+          insertText: `# Google Drive Sample
+type user
+type group
+  relations
+    define member as self
+type folder
+  relations
+    define owner as self
+    define parent as self
+    define viewer as self or owner or viewer from parent
+    define can_create_file as owner
+type doc
+  relations
+    define owner as self
+    define parent as self
+    define viewer as self
+    define can_change_owner as owner
+    define can_share as owner or owner from parent
+    define can_write as owner or owner from parent
+    define can_read as viewer or owner or viewer from parent`,
+          range,
+        },
+        {
+          label: "sample-expenses",
+          kind: monaco.languages.CompletionItemKind.Keyword,
+          insertText: `# Expenses Sample
+type employee
+  relations
+    define manager as self
+    define can_manage as manager or can_manage from manager
+type report
+  relations
+    define submitter as self
+    define approver as can_manage from submitter`,
+          range,
+        },
+        {
+          label: "sample-github",
+          kind: monaco.languages.CompletionItemKind.Keyword,
+          insertText: `# GitHub Sample
+type user
+type organization
+  relations
+    define owner as self
+    define member as self or owner
+    define repo_admin as self
+    define repo_reader as self
+    define repo_writer as self
+type team
+  relations
+    define member as self
+type repo
+  relations
+    define owner as self
+    define admin as self or repo_admin from owner
+    define maintainer as self or admin
+    define writer as self or maintainer or repo_writer from own
+    define triager as self or writerer
+    define reader as self or triager or repo_reader from owner`,
+          range,
+        },
+      ],
+    };
+  };
+
+const provideCompletionItemsOneDotOne =
+  (monaco: typeof MonacoEditor) =>
+  (model: editor.ITextModel, position: Position): languages.CompletionList => {
+    const word = model.getWordUntilPosition(position);
+    const range = {
+      startLineNumber: position.lineNumber,
+      endLineNumber: position.lineNumber,
+      startColumn: word.startColumn,
+      endColumn: word.endColumn,
+    };
+
+    if (position.column === 2) {
```

Comment:
Similar to above, should we use switch/case?

Reply:
Same as above